### PR TITLE
tar: Add basic tar command and package

### PR DESCRIPTION
**Pull request by rjoleary**
Signed-off-by: Ryan O'Leary <ryanoleary@google.com>

### CODE REVIEW

**Comment by rjoleary** [comment]
Tests are still a WIP.

File: cmds/tar/tar.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import (
+	"log"
+	"os"
+
+	flag "github.com/spf13/pflag"
+	"github.com/u-root/u-root/pkg/tar"
+)
+
+var (
+	create  = flag.BoolP("create", "c", false, "create a new tar archive from the given directory")
```
Comment (rminnich) [inline cmds/tar/tar.go]: you're gonna need a t command. There is no escape.

File: cmds/tar/tar.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+func main() {
+	flag.Parse()
+	if flag.NArg() != 1 {
+		flag.Usage()
+		os.Exit(1)
+	}
+
+	if !*create == !*extract {
+		log.Fatal("either supply -c or -x")
```
Comment (rminnich) [inline cmds/tar/tar.go]: I think you should indicate a difference between both being supplied and neither being supplied?

File: cmds/tar/tar.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	extract = flag.BoolP("extract", "x", false, "extract a tar archive from the given directory")
+	verbose = flag.BoolP("verbose", "v", false, "print each filename")
+	file    = flag.StringP("file", "f", "", "tar file")
+)
+
+func main() {
+	flag.Parse()
+	if flag.NArg() != 1 {
+		flag.Usage()
+		os.Exit(1)
```
Comment (rminnich) [inline cmds/tar/tar.go]: could you take more than one? I doubt it's that much harder.

File: cmds/tar/tar.go [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+	}
+
+	if !*create == !*extract {
+		log.Fatal("either supply -c or -x")
+	}
+
+	if *file == "" {
```
Comment (rminnich) [inline cmds/tar/tar.go]: could you just use stdin/stdout if no file supplied?

File: cmds/tar/tar.go (hunk 1 above)
Comment (rjoleary) [reply cmds/tar/tar.go]: I'll address this later in #1168

File: cmds/tar/tar.go (hunk 2 above)
Comment (rjoleary) [reply cmds/tar/tar.go]: Done

File: cmds/tar/tar.go (hunk 3 above)
Comment (rjoleary) [reply cmds/tar/tar.go]: See #1168

File: cmds/tar/tar.go (hunk 4 above)
Comment (rjoleary) [reply cmds/tar/tar.go]: See #1168

**Pull request merged**